esx_service/utils/auth.py

```diff
@@ -84,7 +149,18 @@ def get_tenant(vm_uuid):
             tenant_name = result[0]
             logging.debug("Found tenant_uuid %s, tenant_name %s", tenant_uuid, tenant_name)
 
-    return None, tenant_uuid, tenant_name
+        return None, tenant_uuid, tenant_name
+    else:
+        error_info, tenant_uuid, tenant_name = get_fallback_tenant()
```

Comment:
These two place are different. For normal tenant, from vm_uuid, we query vms table to find the tenant_uuid of the tenant which own this VM. Then we query the tenant table by id to get tenant_name.

For FALLBACK tenant, we know the tenant name is  "__FALLBACK". We then query the tenant table by name to get tenant_uuid.